app/controllers/revision_feedback_controller.rb

```diff
@@ -3,19 +3,54 @@
 require "#{Rails.root}/lib/importers/revision_score_importer"
 
 class RevisionFeedbackController < ApplicationController
+  before_action :initialize_ores
+
   def index
-    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
-    @revision = Revision.find_by(mw_rev_id: params[:rev_id], wiki_id: @wiki.id)
-    set_new_revision unless @revision
-    @feedback = RevisionFeedbackService.new(@revision).feedback
+    set_revision_id
+    unless @revId.nil?
+      features = ores_features(@revId)
+      @feedback = RevisionFeedbackService.new(features).feedback
+    end
   end
 
   private
 
-  def set_new_revision
-    @revision = Revision.new(wiki_id: @wiki.id, mw_rev_id: params[:rev_id], mw_page_id: 0)
-    @revision.save!
-    RevisionScoreImporter.new.update_revision_scores([@revision])
-    @revision.reload
+  def initialize_ores
+    @wiki = Wiki.find_by(language: 'en', project: 'wikipedia')
+    @ores_api = OresApi.new(@wiki)
+  end
+
+    def ores_features(revId)
```

Comment:
RevisionScoreImporter is used in the update cycle to pull in data on all the English Wikipedia mainspace and userspace revisions that don't have score data already. That's used for the recent activity page: https://dashboard.wikiedu.org/recent-activity/dyk

You could maybe extract the code that deals with parsing the ORES output and getting the scores. ArticleRatingExtractor does that for the page assessments; you could similarly make an OresDataExtractor.